internal/app/api/handler/login/login.go

```diff
@@ -50,3 +65,101 @@ func (h *LoginHandler) GetUser(w http.ResponseWriter, r *http.Request) {
 	respond.JSON(w, http.StatusAccepted, u)
 	return
 }
+func (h *LoginHandler) Register(w http.ResponseWriter, r *http.Request) {
+	username := r.FormValue("username")
+	rawPassword := r.FormValue("password")
+	fullName := r.FormValue("fullname")
+	email := r.FormValue("email")
+	deliveryCenter := r.FormValue("deliverycenter")
+	dcs := []string{deliveryCenter}
+	bytes, err := bcrypt.GenerateFromPassword([]byte(rawPassword), 14)
+	if err != nil {
+		h.logger.Errorf("password encryption is fail err: %v", err)
+		respond.JSON(w, http.StatusUnauthorized, "register fail")
+		return
+	}
+	password := string(bytes)
+	user := types.User{
+		ID:             bson.NewObjectId(),
+		Username:       username,
+		FullName:       fullName,
+		Email:          email,
+		DeliveryCenter: dcs,
+		Password:       password,
+	}
+	err = h.srv.AddUser(&user)
+	if err != nil {
+		h.logger.Infof("can't save new user into database err: %v ", err)
+		respond.JSON(w, http.StatusInternalServerError, "add user fail")
+		return
+	}
+	respond.JSON(w, http.StatusCreated, "save user successfully")
+}
+func (h *LoginHandler) ChangeLocalPassword(w http.ResponseWriter, r *http.Request) {
+	username := r.FormValue("username")
+	password := r.FormValue("password")
+	newPassword := r.FormValue("newpassword")
+	_, err := h.loginSvr[1].Authenticate(username, password)
```

Comment:
The logic should be handled by service layer..not here.

Why login service [1] is used here? So if I don't have LDAP account I cannot change my password? And access direct index here is very wrong...If you want to be specific, don't pass a list and access as [0], [1] here.